core/org.eclipse.cdt.core.tests/model/org/eclipse/cdt/core/build/ICBuildConfigurationProviderTests.java

```diff
@@ -0,0 +1,130 @@
+		when(mockToolchain.getBuildConfigNameFragment()).thenReturn("buildConfigName");
+	}
+
+	/**
+	 * Tests org.eclipse.cdt.core.build.ICBuildConfigurationProvider.getCBuildConfigName(IProject, String, IToolChain, String, ILaunchTarget)
+	 * can be extended to provide a customized ICBuildConfiguration name and thereby a customized build output directory name.
+	 *
+	 * @see {@link ICBuildConfigurationProviderTestsProvider#getCBuildConfigName(IProject, String, IToolChain, String, org.eclipse.launchbar.core.target.ILaunchTarget)}
+	 */
+	@Test
+	public void getCBuildConfigName() throws Exception {
+		/*
+		 * Test summary:
+		 * 1) A ICBuildConfiguration is created using the provided toolchain, launch mode and launch target.
+		 * Note, the key principle to the ICBuildConfigurationProvider being overriden successfully is the project nature.
+		 * It uses a customized nature, declared in plugin.xml.
+		 * When the project provider is accessed, the declaration of extension point="org.eclipse.cdt.core.buildConfigProvider" relates
+		 * the customized nature, so the correct (ICBuildConfigurationProviderTestsProvider) provider is used.
+		 *
+		 * 2) ICBuildConfiguration2.getBuildDirectoryURI() queries the build output directory name. The correct value is specified
+		 * by ICBuildConfigurationProviderTestsProvider.getCBuildConfigName.
+		 */
+
+		/*
+		 * cmake because we extend CMakeBuildConfigurationProvider.
+		 * run because ILaunchManager.RUN_MODE.
+		 * buildConfigName because mockToolchain returns this from getBuildConfigNameFragment.
+		 * Local because using Local launch target.
+		 * customizedTest because ICBuildConfigurationProviderTestsProvider.getCBuildConfigName specifies this.
+		 */
+		final String expectedName = "cmake.run.buildConfigName.Local.customizedTest";
+
+		/*
+		 * (1) Create ICBuildConfiguration, using our customized ICBuildConfigurationProviderTestsProvider
+		 */
+		ICBuildConfiguration cBuildConfiguration = configManager.getBuildConfiguration(project, mockToolchain,
+				ILaunchManager.RUN_MODE, launchTargetManager.getLocalLaunchTarget(), new NullProgressMonitor());
+		assertThat(cBuildConfiguration, is(notNullValue()));
+
+		assertThat(cBuildConfiguration, instanceOf(CBuildConfiguration.class));
+		if (cBuildConfiguration instanceof CBuildConfiguration cbc) {
+			/*
+			 * Note, this extra test uses getName on CBuildConfiguration rather than the API ICBuildConfiguration
+			 */
+			String name = cbc.getName();
+			assertThat(name, is(expectedName));
+		}
+
+		assertThat(cBuildConfiguration, instanceOf(ICBuildConfiguration2.class));
+		if (cBuildConfiguration instanceof ICBuildConfiguration2 cbc2) {
+			/*
+			 * (2) Check last segment of build output directory name is expected.
+			 */
+			URI buildDirectoryURI = cbc2.getBuildDirectoryURI();
+			Path path = Paths.get(buildDirectoryURI);
+			String lastSegment = path.getFileName().toString();
+			assertThat(lastSegment, is(expectedName));
+		}
```

Comment:
You can simplify this by removing the extra null and instanceof checks. In tests you can just let NPE and CCE happen without having to do the extra checks for it. 

```suggestion
		CBuildConfiguration cBuildConfig = (CBuildConfiguration) cBuildConfiguration;
		assertThat(cBuildConfig.getName(), is(expectedName));
		/*
		 * (2) Check last segment of build output directory name is expected.
		 */
		Path buildDirectory = cBuildConfig.getBuildDirectory();
		String lastSegment = buildDirectory.getFileName().toString();
		assertThat(lastSegment, is(expectedName));
```

Reply:
Great.
That's simpler and more concise then.
Also I like cBuildConfig.getBuildDirectory(); rather than cbc2.getBuildDirectoryURI();